pystac/extensions/raster.py

```diff
@@ -741,6 +746,42 @@ def summaries(
         return SummariesRasterExtension(obj)
 
 
+class AssetRasterExtension(RasterExtension[pystac.Asset]):
+    asset_href: str
+    """The ``href`` value of the :class:`~pystac.Asset` being extended."""
+
+    properties: Dict[str, Any]
+    """The :class:`~pystac.Asset` fields, including extension properties."""
+
+    additional_read_properties: Optional[Iterable[Dict[str, Any]]] = None
+    """If present, this will be a list containing 1 dictionary representing the
+    properties of the owning :class:`~pystac.Item`."""
+
+    def __init__(self, asset: pystac.Asset):
+        self.asset_href = asset.href
+        self.properties = asset.extra_fields
+        if asset.owner and isinstance(asset.owner, pystac.Item):
+            self.additional_read_properties = [asset.owner.properties]
+
+    def __repr__(self) -> str:
+        return "<AssetRasterExtension Asset href={}>".format(self.asset_href)
+
+
+class ItemAssetsRasterExtension(RasterExtension[item_assets.AssetDefinition]):
+    asset_defn: item_assets.AssetDefinition
```

Comment:
Nit, letters are cheap and abbreviations hide meaning:

```suggestion
    asset_definition: item_assets.AssetDefinition
```